### PR TITLE
Update API Blueprint

### LINKED ISSUE #249: Compare API blueprint documentation to actual output
We want to compare the API blueprint document we've generated to the actual output and fix any inconsistencies on the documentation side.

http://docs.codecorpsapi.apiary.io

### PR DESCRIPTION
**Pull request by joshsmith**
Closes #249 
- Update JWT exchange
- Add JWT refresh
- Add JWT expired response
- Reorder tasks
- Remove all links
- Update 422 error
- Remove comment user mentions
- Refine category resource grouping
- Fix 404 response
- Remove 409 responses
- Fix tasks response

...and a whole host of innumerable other changes and additions.

### CODE REVIEW

File: blueprint/api.apib [hunk 1]
```diff
@@ -726,114 +854,207 @@ A user is able to have multiple roles through their user roles.
...
+
+    + Attributes (Role Response)
+
++ Response 404 (application/vnd.api+json; charset=utf-8)
+
+    + Attributes (Record Not Found Response)
+
+# Group Role Skills
+
+This resource identifies a relationship between a Role and a Skill. For example, the skill of "JavaScript" may relate to the role of "Backend Developer," and also "Front End Developer" and even "Mobile Developer".
```
Comment (rileytaylor) [inline blueprint/api.apib]: This is a little awkward. It almost sounds like the relationships are variable in suggestion. I get the reason, but maybe just list them out? 

> ... role of "Backend", "Frontend" and "Mobile"

This is more of an opinion, so take it as you will.

**Review by rileytaylor — APPROVED** [review]
Looks good. Definitely squash and rebase of course. One opinionated comment, take it as you will.

File: blueprint/api.apib [hunk 2]
```diff
@@ -34,22 +34,48 @@ Exchange email and password for `token`.
...
+
++ Response 401 (application/vnd.api+json; charset=utf-8)
+
+    + Attributes (JSON Web Token Expired Response)
+
 # Group Categories
 
+This endpoint is for creating and returning the categories available for projects and users.
+
+The Category resource allows us to categorize projects (Project Category), as well as identify a user's interests (User Category). The catgories of "Society", "Technology" or "Government", for example, could all be used to categorize projects or match to a user's given interests.
```
Comment (begedin) [inline blueprint/api.apib]: This line has a typo. `catgories` -> `categories`

File: blueprint/api.apib [hunk 3]
```diff
@@ -108,128 +131,157 @@ Exchange email and password for `token`.
...
-            Authorization: Bearer {token}
+            Accept: application/vnd.api+json
 
-+ Response 200 (application/json)
++ Response 200 (application/vnd.api+json; charset=utf-8)
 
     + Attributes (Comments Response)
 
-## List comments for a specified task [GET /tasks/{number}/comments]
+## Task Comment [/tasks/{task_id}/tasks/{id}]
```
Comment (begedin) [inline blueprint/api.apib]: `[/tasks/{task_id}/comments/{id}]`, i'm assuming

File: blueprint/api.apib [hunk 4]
```diff
@@ -726,114 +854,207 @@ A user is able to have multiple roles through their user roles.
...
+
+    + Attributes (JSON Web Token Invalid Response)
+
++ Response 403 (application/vnd.api+json; charset=utf-8)
+
+    + Attributes (Forbidden Response)
+
+# Group Skills
+
+This endpoint retrieves Skills that can belong to Projects, Roles, or Users. This helps to match users to relevant projects or roles.
```
Comment (begedin) [inline blueprint/api.apib]: Saying it belongs to would imply it's a one to many relationship. I'd say go with "that can relate to", or if that's not enough, be more verbose about those relations.

Example of more verbosity:

```
Skills relate to projects in the context of "some of these skills are needed for someone to work on this project".

They relate to roles in the context of "to fulfil a role, a user needs to possess one or more skills. Having a skill means they can fulfil one or more roles."

They relate to users in the context that a user can have one or more skills. Multiple users can have the same skills.
```

**Review by begedin — CHANGES_REQUESTED** [review]
Couple of minor issues I've found.

**Pull request merged**